src/Client/index.ts

```diff
@@ -263,24 +267,44 @@ export class Client {
     return `${address}:${port}`;
   };
 
-  protected metadata = ({
-    credentials = this.#defaultCredentials,
-    requiresLeader,
-  }: BaseOptions): Metadata => {
+  private createCredentialsMetadataGenerator = ({
+    username,
+    password,
+  }: Credentials): Parameters<
```

Comment:
**SUGGESTION:** I know that this is a valid TS syntax, but could we introduce some typing to wrap this type definition to make it less cryptic?

Reply:
Unfortunately the type isnt exported. I considered a type alias at the top of the file, but I dont think it makes it any clearer, it just adds an extra step.

Reply:
I think that alias on top of the file would be good. I believe that this would help us with future maintenance.